plugins/rollbar_dashboards/lib/rollbar_dashboards/client.rb

```diff
@@ -3,83 +3,65 @@
-      end
-
-      if response.success?
-        data = ::JSON.parse(response.body, symbolize_names: true)
+      request :get, "#{@base}/reports/top_active_items?#{args}" do |data|
         data.fetch(:result).map { |item_hash| item_hash.fetch(:item) }
       end
-    rescue StandardError
-      Samson::ErrorNotifier.notify($!)
-      nil
     end
 
     # creates a new rql job
     def create_rql_job(query_string)
-      args = {
+      params = {
         access_token: @token,
         query_string: query_string,
         force_refresh: 1
       }
-
-      response = Faraday.post("#{@base}/rql/jobs", args) do |req|
-        req.options.timeout = req.options.open_timeout = REQUEST_TIMEOUT
-      end
-
-      if response.success?
-        Rails.logger.info "Created RQL job: #{response.body}"
-        ::JSON.parse(response.body, symbolize_names: true).dig_fetch(:result, :id)
+      request :post, "#{@base}/rql/jobs", params do |data|
+        data.dig_fetch(:result, :id)
       end
-    rescue StandardError
-      Rails.logger.error "Failed to create RQL job"
-      Samson::ErrorNotifier.notify($!)
-      nil
     end
 
     # gets the rql job result, returns array of hashes e.g. [{ col1: row_val_1 ...}, { col1: row_val_2 ...}]
     def rql_job_result(id)
       MAX_RQL_JOB_WAIT_TIME.times do |i|
-        response = Faraday.get("#{@base}/rql/job/#{id}/result?access_token=#{@token}") do |req|
-          req.options.timeout = req.options.open_timeout = REQUEST_TIMEOUT
+        request :get, "#{@base}/rql/job/#{id}/result?access_token=#{@token}" do |data|
+          if data.dig_fetch(:result, :result).nil?
+            Rails.logger.info "Waiting for RQL job #{id}"
+            sleep 1 if i < (MAX_RQL_JOB_WAIT_TIME - 1)
+          else
+            columns = data.dig_fetch(:result, :result, :columns).map(&:to_sym)
+            rows = data.dig_fetch(:result, :result, :rows)
+            return rows.map { |row| columns.zip(row).to_h }.presence
+          end
         end
+      end
 
-        break unless response.success?
-
-        data = ::JSON.parse(response.body, symbolize_names: true)
-
-        if data.dig_fetch(:result, :result).nil?
-          Rails.logger.info "Waiting for RQL job #{id}"
-          sleep(1) if i < (MAX_RQL_JOB_WAIT_TIME - 1)
-        else
-          columns = data.dig_fetch(:result, :result, :columns).map(&:to_sym)
-          rows = data.dig_fetch(:result, :result, :rows)
+      raise Samson::Hooks::UserError, "Timeout retrieving RQL job #{id} result"
+    end
 
-          return rows.map { |row| columns.zip(row).to_h }.presence
-        end
+    def request(method, url, *args)
+      response = Faraday.send(method, url, *args) do |req|
+        req.options.timeout = req.options.open_timeout = REQUEST_TIMEOUT
       end
-
-      nil
+      raise "Response #{response.status}" unless response.success?
+      yield JSON.parse(response.body, symbolize_names: true)
```

Comment:
worth using faraday json middleware instead of custom code?

Reply:
2 lines of code vs indirection/config, seems kinda the same to me 🤷‍♂